src/custom-pv-components/pdb-navigation.js

```diff
@@ -17,25 +17,58 @@ const height = 40,
   };
 
 class ProtvistaPdbNavigation extends ProtvistaNavigation {
-  
+
   connectedCallback() {
     this.style.display = 'block';
     this.style.width = '100%';
     this.width = this.offsetWidth;
 
     this._offset = parseFloat(this.getAttribute('offset')) || 0;
+    this._length = parseInt(this.getAttribute('length'));
+    this._displaystart = parseInt(this.getAttribute('displaystart')) || (this._offset > 0) ? this._offset : 1;
+    this._displayend = parseInt(this.getAttribute('displayend')) || (this._offset > 0) ? (this._length + this._offset - 1) : this._length;
+    this._highlightstart = parseInt(this.getAttribute('highlightstart'));
+    this._highlightend = parseInt(this.getAttribute('highlightend'));
+    this._highlightintervals = this.getAttribute("highlightintervals");
 
-    this._length = parseFloat(this.getAttribute('length'));
-    this._displaystart = parseFloat(this.getAttribute('displaystart')) || (this._offset > 0) ? this._offset : 1;
-    this._displayend = parseFloat(this.getAttribute('displayend')) || (this._offset > 0) ? (this._length + this._offset - 1) : this._length;
-    this._highlightStart = parseFloat(this.getAttribute('highlightStart'));
-    this._highlightEnd = parseFloat(this.getAttribute('highlightEnd'));
+    this._highlightedFragments = [];
 
     this._onResize = this._onResize.bind(this);
+    this._updateNavRuler = this._updateNavRuler.bind(this);
 
     this._createNavRuler();
   }
 
+  static get observedAttributes() {
+    return ['length', 'displaystart', 'displayend', 'highlightstart', 'highlightend', 'width', 'highlightintervals'];
+  }
+
+  attributeChangedCallback(name, oldValue, newValue) {
+    if (oldValue !== newValue) {
+      if (name === "highlightintervals") this[`_${name}`] = newValue;
+      else if (name === "offset") this[`_${name}`] = parseFloat(newValue);
+      else this[`_${name}`] = parseInt(newValue);
```

Comment:
[subjective] if/else if/else blocks with `{}` are harder to read than using the brackets?

Reply:
Is more about im used to if blocks without brackets when is only one statment... 😝 but I understand it could be more readable with the indentation lvl with `{}`